tensorrt_llm/_torch/custom_ops/cpp_custom_ops.py

```
@@ -238,6 +238,313 @@ def _(
+            _check_turboquant4_same_static_dim(
+                scales.shape[dim], cache.shape[dim],
+                "TurboQuant4 cache/scales leading shapes mismatch.")
+        if isinstance(scales.shape[4], int) and scales.shape[4] != 1:
+            raise RuntimeError("TurboQuant4 scale cache last dimension must be 1.")
+        if require_kv and isinstance(cache.shape[1],
+                                     int) and cache.shape[1] < 2:
+            raise RuntimeError(
+                "TurboQuant4 attention requires key and value cache entries.")
+
+    @torch.library.register_fake("trtllm::turboquant4_quantize")
+    def _(input: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
```

Ruff (0.15.13)

[error] Function argument `input` is shadowing a Python builtin

(A002)